docs/OV_Runtime_UG/Python_API_inference.md

```diff
@@ -0,0 +1,60 @@
+# OpenVINO™ Runtime Python API Advanced Inference {#openvino_docs_OV_UG_Python_API_inference}
+
+
+.. warning:: All mentioned methods are very dependant on a specific hardware and software set-up. Please consider conducting your own experiments with various models and different input/output sizes. Presented methods are not universal, they may or may not apply to the specific pipeline. Please consider all tradeoffs and avoid premature optimizations. 
+
+
+Direct Inference with ``CompiledModel``
+#######################################
+
+``CompiledModel`` class provides ``__call__`` method which runs a single synchronous inference using given model. In addition to a compact code, all future calls to ``CompiledModel.__call__`` result in less overhead as the object reuse already created ``InferRequest``.
+
+
+.. doxygensnippet:: docs/snippets/ov_python_inference.py
+   :language: python
+   :fragment: [direct_inference]
+
+
+Shared Memory on Inputs
+#######################
+
+While using ``CompiledModel``, ``InferRequest`` and ``AsyncInferQueue`` inference methods, OpenVINO™ Runtime Python API provides additional "Shared Memory" mode. Specify ``shared_memory`` flag to enable or disable this feature. "Shared Memory" mode could be beneficial when inputs are large and copying of data is considered an expensive operation. This feature creates shared ``Tensor`` instances with "zero-copy" approach, reducing overhead of setting inputs to minimum. Example usage:
+
+
+.. doxygensnippet:: docs/snippets/ov_python_inference.py
+   :language: python
+   :fragment: [shared_memory_inference]
+
+
+.. note:: "Shared Memory" is enabled by default in ``CompiledModel.__call__``. For other methods like ``InferRequest.infer`` or ``InferRequest.start_async``, it is required to manually set the flag to ``True``.
+
+
+.. warning:: When data is being shared, all modifications may affect inputs of the inference! Use this feature with caution, especially in multi-threaded/parallel code where data can be modified outside of the function's control flow.
```

Comment:
I would leave it as it is. In general shared memory could be considered "problematic". For example, numpy is well known to behave differently for copies and for views of data. It is not common knowledge at the entry-level. I know (even personally, good old days of university:)) there are a lot of mistakes because of this concept.

Putting extra emphasis on multi-threading only helps in such case.

Reply:
In addition, this "automatic sharing" is due to simplicity of calls from `CompiledModel` usually it will be used in sequential pipeline as there is nothing else returned, just `OVDict`.

It is more complicated when it comes down to rich interface of `InferRequest` and it's ability to chain models, reuse Tensors and start asyncs. There applying shared mode by default was considered risky and set to `False` to prevent issues (and keep backward compatibility).